plugins/repository-hdfs/build.gradle

```diff
@@ -7,427 +7,388 @@
  */
 
 import org.apache.tools.ant.filters.ReplaceTokens
-import org.apache.tools.ant.taskdefs.condition.Os
 import org.elasticsearch.gradle.internal.info.BuildParams
 import org.elasticsearch.gradle.internal.test.RestIntegTestTask
-
-import java.nio.file.Files
+import org.elasticsearch.gradle.internal.util.HdfsUtils
 import java.nio.file.Path
-import java.nio.file.Paths
 
 import static org.elasticsearch.gradle.PropertyNormalization.IGNORE_VALUE
 
 apply plugin: 'elasticsearch.test.fixtures'
-apply plugin: 'elasticsearch.rest-resources'
-apply plugin: 'elasticsearch.rest-test'
+apply plugin: 'elasticsearch.internal-java-rest-test'
+apply plugin: 'elasticsearch.internal-yaml-rest-test'
 
 esplugin {
-  description 'The HDFS repository plugin adds support for Hadoop Distributed File-System (HDFS) repositories.'
-  classname 'org.elasticsearch.repositories.hdfs.HdfsPlugin'
+    description 'The HDFS repository plugin adds support for Hadoop Distributed File-System (HDFS) repositories.'
+    classname 'org.elasticsearch.repositories.hdfs.HdfsPlugin'
 }
 
 versions << [
-  'hadoop': '3.3.1'
+        'hadoop': '3.3.1'
 ]
 
 final int minTestedHadoopVersion = 2;
 final int maxTestedHadoopVersion = 3;
 
 testFixtures.useFixture ":test:fixtures:krb5kdc-fixture", "hdfs"
 
-for (int hadoopVersion = minTestedHadoopVersion; hadoopVersion <= maxTestedHadoopVersion; hadoopVersion++) {
-  configurations.create("hdfs" + hadoopVersion + "Fixture")
+configurations {
+    krb5Config
+    krb5Keytabs
 }
-
 dependencies {
-  api project(path: 'hadoop-client-api', configuration: 'shadow')
-  if (isEclipse) {
-    /*
-     * Eclipse can't pick up the shadow dependency so we point it at *something*
-     * so it can compile things.
-     */
-    api project(path: 'hadoop-client-api')
-  }
-  runtimeOnly "org.apache.hadoop:hadoop-client-runtime:${versions.hadoop}"
-  implementation "org.apache.hadoop:hadoop-hdfs:${versions.hadoop}"
-  api 'org.apache.htrace:htrace-core4:4.0.1-incubating'
-  runtimeOnly 'com.google.guava:guava:27.1-jre'
-  api 'com.google.protobuf:protobuf-java:2.5.0'
-  api 'commons-logging:commons-logging:1.1.3'
-  api "org.apache.logging.log4j:log4j-1.2-api:${versions.log4j}"
-  api 'commons-cli:commons-cli:1.2'
-  api "commons-codec:commons-codec:${versions.commonscodec}"
-  api 'commons-collections:commons-collections:3.2.2'
-  api 'org.apache.commons:commons-configuration2:2.7'
-  api 'commons-io:commons-io:2.4'
-  api 'commons-lang:commons-lang:2.6'
-  api 'javax.servlet:servlet-api:2.5'
-  api "org.slf4j:slf4j-api:${versions.slf4j}"
-  api "org.apache.logging.log4j:log4j-slf4j-impl:${versions.log4j}"
-  // Set the keytab files in the classpath so that we can access them from test code without the security manager
-  // freaking out.
-  if (isEclipse == false) {
-    testRuntimeOnly files(project(':test:fixtures:krb5kdc-fixture').ext.krb5Keytabs("hdfs", "hdfs_hdfs.build.elastic.co.keytab").parent) {
```

Comment:
Do not use this pattern of `project(':test:fixtures:krb5kdc-fixture').ext.krb5Keytabs` etc for resolve files from other projects. Resolve from outgoing dependencies instead